assets/src/blocks/CarouselHeader/Sidebar.js

```diff
@@ -1,43 +1,153 @@
+import { useState, useRef, useEffect } from 'react';
 import {
   CheckboxControl,
   PanelBody,
-  PanelRow,
 } from '@wordpress/components';
-import { Fragment } from '@wordpress/element';
-
 import { InspectorControls } from '@wordpress/block-editor';
 import { SidebarSlide } from './SidebarSlide';
-import { useState, useRef } from 'react';
 
 const { __ } = wp.i18n;
 
-const reOrderSlides = (slides, oldIndex, newIndex) => {
-  const movingSlide = slides[oldIndex];
-  return slides.reduce((orderedSlides, slide, index) => {
-    if (index === newIndex) {
-      orderedSlides.push(movingSlide);
-    }
-    if (index !== oldIndex) {
-      orderedSlides.push(slide);
-    }
-    if (index === slides.length -1 && newIndex > index) {
-      orderedSlides.push(movingSlide);
-    }
-    return orderedSlides;
-  }, []);
-}
-
 export const Sidebar = ({
   carouselAutoplay,
   setAttributes,
   slides,
-  currentSlide,
   changeSlideAttribute,
   goToSlide,
 }) => {
   const [dragTarget, setDragTarget] = useState(null);
   const [draggedSlide, setDraggedSlide] = useState(null);
-  const timeoutRef = useRef(null);
+  const slidesRef = useRef(null);
+
+  const getNode = (index) => (
```

Comment:
This function can be defined outside of render.

Reply:
Do you mean `render` as a "scope"? Because this function is not being created through every render method.

Reply:
I understand that you're referring to move out of the component, am I right?

Reply:
Yes, I meant render as in "when the component renders".